tests/components/yessssms/test_notify.py

```diff
@@ -1,7 +1,150 @@
 """The tests for the notify yessssms platform."""
 import unittest
+from unittest.mock import patch
+
+import pytest
 import requests_mock
+
+from homeassistant.setup import async_setup_component
 import homeassistant.components.yessssms.notify as yessssms
+from homeassistant.components.yessssms.const import CONF_PROVIDER
+
+from homeassistant.const import CONF_PASSWORD, CONF_RECIPIENT, CONF_USERNAME
+
+
+@pytest.fixture(name="config")
+def config_data():
+    """Set valid config data."""
+    config = {
+        "notify": {
+            "platform": "yessssms",
+            "name": "sms",
+            CONF_USERNAME: "06641234567",
+            CONF_PASSWORD: "secretPassword",
+            CONF_RECIPIENT: "06509876543",
+            CONF_PROVIDER: "educom",
+        }
+    }
+    return config
+
+
+@pytest.fixture(name="valid_settings")
+def init_valid_settings(hass, config):
+    """Initialize component with valid settings."""
+    return async_setup_component(hass, "notify", config)
+
+
+@pytest.fixture(name="invalid_provider_settings")
+def init_invalid_provider_settings(hass, config):
+    """Set invalid provider data and initalize component."""
+    config["notify"][CONF_PROVIDER] = "FantasyMobile"  # invalid provider
+    return async_setup_component(hass, "notify", config)
+
+
+@pytest.fixture(name="invalid_login_data")
+def mock_invalid_login_data():
+    """Mock invalid login data."""
+    path = "homeassistant.components.yessssms." "notify.YesssSMS.login_data_valid"
```

Comment:
Please remove the separation in the string. It's a side effect after black formatting.